app/javascript/app/utils/user-validation.js

```diff
@@ -0,0 +1,30 @@
+export default user => {
+  const allFieldsFilled = Object.keys(user).every(k => user[k]);
+  const passConfirmationIsOk = user.password === user.password_confirmation;
+  const emailregex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  const emailValid = user.email && user.email.match(emailregex);
```

Comment:
I'd move it to another file `email-validation` and `password-validation` that would implement `validate` function and use it like so:
```
import emailValidator from 'email-validation';

const isEmailValid = emailValidator.validate(user.email)
```
also wdyt about changing the var names to more readable, like isEmailValid, isPasswordValid, etc?